From 7a24bda57c62b6b7c90035fc8f0963d08ec8d8f4 Mon Sep 17 00:00:00 2001
From: Jiping Ma <jiping.ma2@windriver.com>
Date: Sun, 17 Mar 2024 23:32:42 -0700
Subject: [PATCH 6/8] iavf: Include xdp.h where required

This commit accommodates commit 680ee0456a57 ("net: invert the netdevice.h
vs xdp.h dependency") which removes the "#include <xdp.h>" C preprocessor
directive from header file "netdevice.h". This commit was merged in the
v6.6 development cycle and causes compilation failures with the v6.6 kernel,
which are listed below.

We accommodate this commit by including the "xdp.h" header file as needed
from "iavf_txrx.h".

References:
* https://git.yoctoproject.org/linux-yocto/commit/?h=680ee0456a57

Resolved compilation errors:

src/iavf_txrx.h:456:22: error: field xdp_rxq has incomplete type
  456 |  struct xdp_rxq_info xdp_rxq;

Signed-off-by: Jiping Ma <jiping.ma2@windriver.com>
---
 src/iavf_txrx.h | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/src/iavf_txrx.h b/src/iavf_txrx.h
index a2a6a22..777b601 100644
--- a/src/iavf_txrx.h
+++ b/src/iavf_txrx.h
@@ -4,6 +4,8 @@
 #ifndef _IAVF_TXRX_H_
 #define _IAVF_TXRX_H_
 
+#include <net/xdp.h>
+
 /* Interrupt Throttling and Rate Limiting Goodies */
 #define IAVF_DEFAULT_IRQ_WORK      256
 
-- 
2.42.0

